src/mon/OSDMonitor.cc

```diff
@@ -592,6 +592,11 @@ int OSDMonitor::reweight_by_utilization(int oload,
     std::pair<int, float> osd_util;
     osd_util.first = p->first;
     if (by_pg) {
+      if (pgs_by_osd[p->first] == 0) {
```

Comment:
Seems we have set the array size at max_osd at 
https://github.com/ceph/ceph/pull/8941/files#diff-0a5db46a44ae9900e226289a810f10e8R500.